misc/odin_filereg.bat

```diff
@@ -0,0 +1,163 @@
+	echo @cd %%~f2>> %ODIN_ROOT%\odin.bat
+	echo odin.exe %%1 .>> %ODIN_ROOT%\odin.bat
+	if %PAUSE_MODE%==onerrors (
+		echo @if %%errorlevel%% neq 0 pause>> %ODIN_ROOT%\odin.bat
+	) else if %PAUSE_MODE%==always (
+		echo @pause>> %ODIN_ROOT%\odin.bat
+	)
+) else if %CALL_MODE%==comspec (
+	if %PAUSE_MODE%==always (
+		set ODIN_CMD="%%comspec%% /c %ODIN_EXE% run \"%%w\"&&pause"
+		set ODIN_FOLDER_CMD="%%comspec%% /c %ODIN_EXE% %%x \"%%1\"&&pause"
+	) else (
+		set ODIN_CMD="%%comspec%% /c %ODIN_EXE% run \"%%w\""
+		set ODIN_FOLDER_CMD="%%comspec%% /c %ODIN_EXE% %%x \"%%1\""
+	)
+) else (
+	set ODIN_CMD="%ODIN_EXE% run \"%%w\""
+	set ODIN_FOLDER_CMD="%ODIN_EXE% %%x \"%%1\""
+)
+@echo on
+reg add "%ODIN_FILE%" /ve /d "Odin source file" %REG_OPT%
+reg add "%ODIN_SOURCE_FILE_EXT%" /ve /d "%ODIN_FILE%" %REG_OPT%
+if %CONTEXT_MENU_FILE%==true (
+	reg add "%ODIN_FILE%\DefaultIcon" /d "%ODIN_EXE%,1" /t REG_EXPAND_SZ %REG_OPT%
+	reg add "%ODIN_FILE%\shell" %REG_OPT%
+	reg add "%ODIN_FILE%\shell\run" /ve /d "Odin Run" %REG_OPT%
+	reg add "%ODIN_FILE%\shell\run\command" /d %ODIN_CMD% /t REG_EXPAND_SZ %REG_OPT%
+)
+if %CONTEXT_MENU_FOLDER%==true (
+	reg add "%ODIN_FOLDER%" /v "MUIVerb" /d "Odin" %REG_OPT%
+	reg add "%ODIN_FOLDER%" /v "subcommands" /d "" %REG_OPT%
+	reg add "%ODIN_FOLDER%" /v "Icon" /d "%ODIN_EXE%,0" /t REG_EXPAND_SZ %REG_OPT%
+	@for %%x in (build,run,check) do (
+		reg add "%ODIN_FOLDER%\Shell\%%x" /d "%%x" %REG_OPT%
+		reg add "%ODIN_FOLDER%\Shell\%%x\command" /d %ODIN_FOLDER_CMD% /t REG_EXPAND_SZ %REG_OPT%
+	)
+)
+@echo off
+goto end_of_script
+
+:remove_file_reg
+@echo on
+reg delete "%ODIN_FOLDER%" %REG_OPT%
+reg delete "%ODIN_SOURCE_FILE_EXT%" %REG_OPT%
+reg delete "%ODIN_FILE%" %REG_OPT%
+@echo off
+goto end_of_script
+
+:show_help
+echo.
+echo Usage: %~n0 [options]
+echo.
+echo options: (*) is default
+echo.
+echo -help             Show help
+echo -add              Add application and file extensions
+echo -remove           Remove application and file extensions
+echo.
+echo -use-batch        Use a batch file to call Odin.exe (*)
+echo -use-exe          Call Odin.exe directly
+echo -use-comspec      Use the defined comspec to call Odin (%comspec%)
+echo.
+echo -pause-always     Always pause (*)
+echo -pause-on-errors  Only pause on error
+echo -pause-never      Never pause
+echo.
+echo -context-none     Register .odin file extentions only
+echo -context-file     Add Odin context menu for .odin file extentions
+echo -context-folder   Add Odin context menu to folders
+echo -context-all      Add both (*)
+echo.
+echo Note: Using a batch file works best with the pause
+echo See:  https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/reg
+echo.
+@pushd %~dp0..
+if not defined ODIN_ROOT (
+    echo Environment variable ODIN_ROOT is NOT defined, set it to the folder path for Odin.exe
+    echo This can be done by executing the following command:
+    echo setx ODIN_ROOT %CD%
+) else (
+    odin.exe report
```

Comment:
if ODIN_ROOT is defined a call to odin.exe report is made this will check if the path for ODIN_ROOT exists